src/pkg/services/m365/api/graph/service_mock.go

```diff
@@ -25,29 +25,31 @@ func NewService(
 		return nil, clues.Wrap(err, "generating graph adapter")
 	}
 
-	return graph.NewService(a), nil
+	return NewService(a), nil
 }
 
-// CreateAdapter is similar to graph.CreateAdapter, but with option to
+// CreateGockAdapter is similar to graph.CreateAdapter, but with option to
 // enable interceptions via gock to make it mockable.
-func CreateAdapter(
+func CreateGockAdapter(
 	tenant, client, secret string,
 	counter *count.Bus,
-	opts ...graph.Option,
-) (*msgraphsdkgo.GraphRequestAdapter, error) {
-	auth, err := graph.GetAuth(tenant, client, secret)
+	opts ...Option,
+) (abstractions.RequestAdapter, error) {
+	auth, err := GetAuth(tenant, client, secret)
 	if err != nil {
 		return nil, err
 	}
 
-	httpClient, _ := graph.KiotaHTTPClient(counter, opts...)
+	httpClient, cc := KiotaHTTPClient(counter, opts...)
 
 	// This makes sure that we are able to intercept any requests via
 	// gock. Only necessary for testing.
 	gock.InterceptClient(httpClient)
 
-	return msgraphsdkgo.NewGraphRequestAdapterWithParseNodeFactoryAndSerializationWriterFactoryAndHttpClient(
+	ng, err := msgraphsdkgo.NewGraphRequestAdapterWithParseNodeFactoryAndSerializationWriterFactoryAndHttpClient(
 		auth,
 		nil, nil,
 		httpClient)
+
+	return wrapAdapter(ng, cc), clues.Stack(err).OrNil()
```

Comment:
just checking, but I take it this is the line that really matters since it should hook into error wrapping from previous PRs?

Reply:
correct!